### PR TITLE
Adding useWrapperFileDirectly Option

### PR DESCRIPTION
**Pull request by ArnoACS**
Adding an option to use legacy wrapper file instead of generate it each time.

### CODE REVIEW

**Comment by olamy** [comment]
LGTM thanks for this!
Do you have a bit of time to add a test as well?

**Comment by ArnoACS** [comment]
Thanks for your review code. 
I added a test on the boolean assignment, but I don't know how to test the impact on index.groovy. 
I hope that's enough.

**Comment by ArnoACS** [comment]
@olamy Hi ! It's not enough to close pull request ?
Did I do something wrong?

File: src/test/java/htmlpublisher/HtmlPublisherIntegrationTest.java [hunk 1]
```diff
@@ -182,6 +182,40 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
...
+                FilePath ws = build.getWorkspace().child(reportDir);
+                ws.child("tab1.html").write("hello", "UTF-8");
+                ws.child("tab2.html").write("hello", "UTF-8");
+                return true;
+            }
+        });
+
+
+        HtmlPublisherTarget target1 = new HtmlPublisherTarget("tab1", reportDir, "tab1.html", true, true, false);
+        //default behavior is to not use wrapper file directly
```
Comment (jglick) [inline src/test/java/htmlpublisher/HtmlPublisherIntegrationTest.java:201]: That seems to contradict https://github.com/jenkinsci/htmlpublisher-plugin/blob/126319b8aefa14601d1893bfecf964f5d37c0b3a/src/main/resources/htmlpublisher/HtmlPublisherTarget/config.jelly#L41

File: src/main/resources/htmlpublisher/HtmlPublisherTarget/help-useWrapperFileDirectly.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div>
+  By default, each archived reports access generates a new wrapper file.
+  If checked, the legacy wrapper file will be used instead of generate it each time.
+  Must only be used if wrapper file was been transformed and saved with checksum
```
Comment (jglick) [inline src/main/resources/htmlpublisher/HtmlPublisherTarget/help-useWrapperFileDirectly.html:4]: What is a “legacy wrapper file”? The term _wrapper_ is not mentioned in https://github.com/jenkinsci/htmlpublisher-plugin/blob/8a37a361a73c34b0ef7de9720b934d03bc0b47a9/README.md at all. As a potential user of the plugin I would have no idea what to make of this help text or whether I should check this box or not.

File: src/main/resources/htmlpublisher/HtmlPublisherTarget/config.jelly [hunk 3]
```diff
@@ -37,5 +37,8 @@
       <f:entry field="escapeUnderscores" title="${%escapeUnderscores.title}">
         <f:checkbox default="true"/>
       </f:entry>
+    <f:entry field="useWrapperFileDirectly" title="${%useWrapperFileDirectly.title}">
+      <f:checkbox default="true"/>
```
Comment (daniel-beck) [inline src/main/resources/htmlpublisher/HtmlPublisherTarget/config.jelly:41]: Why would a legacy option be enabled by default?

**Review by daniel-beck — COMMENTED** [review]
Caused https://www.jenkins.io/security/advisory/2026-04-29/#SECURITY-3706

**Pull request merged**